ServicesWrappersJava/src/main/java/com/careerbuilder/datascienceservices/ResponseBase.java

```
@@ -0,0 +1,81 @@
+package com.careerbuilder.datascienceservices;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public abstract class ResponseBase {
+
+	protected String error;
+
+	protected JSONObject jsonObject;
+
+	protected void processHTTPResponse(HttpResponse httpresp)
+			throws IOException, Exception {
+
+		boolean isError = ParseError(httpresp);
+
+		if (!isError) {
+
+			jsonObject = getJSONObject(httpresp);
+
+			parseJSONObject();
+		}
+
+	}
+
+	protected boolean ParseError(HttpResponse data) throws IOException,
+			Exception {
+
+		if (data.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
+
+			String responseData = parseErrorResponse(data);
+
+			this.error = data.getStatusLine().toString() + " " + responseData;
+
+			return true;
+
+		} else {
+
+			return false;
+		}
+	}
+	
+	private static String parseErrorResponse(HttpResponse data)
+			throws Exception, IOException {
```

ditto from above re: redundant exception declarations -- and do you really need to throw any exceptions here? is there something sensible you could have this method do if it incurs an exception while trying to parse an error string?